### PR TITLE
Healthz sync error message

### LINKED ISSUE #9025: /healthz fails when eth1 stalls
# 🐞 Bug Report

### Description

When eth1 network stalls (see goerli), the `/healthz` endpoint starts to return an error.
```
*powchain.Service: ERROR, eth1 node is still syncing
```

Many deployments consider `/healthz` as an indicator on whether or not a Prysm node could serve traffic. However, a Prysm node could actually serve traffic when eth1 is stalled. The risk in mainnet is that if eth1 mainnet ever stalled entirely, then we could see many eth2 prysm validators stop participating entirely. 

If there was some way to understand whether or not eth1 is stalled or if the eth1 node is out of sync, then we could simply report an error only on the latter. However, there may still be a risk here if many nodes suddenly consider themselves out of sync on mainnet. 

Suggestion: Return error when eth1 is out of sync AND prysm does not have a genesis state. Additionally, surface some kind of notification to the user that their eth1 client is out of sync / syncing and proposals may fail to include deposits as a result. 

### Has this worked before in a previous version?

N/A

## 🔬 Minimal Reproduction

Have an eth1 endpoint that stops advancing. 

## 🔥 Error

<pre><code>*powchain.Service: ERROR, eth1 node is still syncing</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
All
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
v1.3.10
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

### PR DESCRIPTION
**Pull request by baxterjfinch**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**
Modifies the /healthz powService error message when geth is stalled and genesis state is missing

**Which issues(s) does this PR fix?**
https://github.com/prysmaticlabs/prysm/issues/9025

Fixes #9025

### CODE REVIEW

File: beacon-chain/powchain/service.go [hunk 1]
```diff
@@ -545,6 +547,12 @@ func (s *Service) waitForConnection() {
 				}).Info("Connected to eth1 proof-of-work chain")
 				return
 			}
+			_, err := s.cfg.BeaconDB.GenesisState(s.ctx)
+			if err != nil {
```
Comment (terencechain) [inline beacon-chain/powchain/service.go:551]: An error here doesn't imply genesis state is not available

**Review by terencechain — CHANGES_REQUESTED** [review]

File: beacon-chain/powchain/service.go (hunk 1 above)
Comment (baxterjfinch) [reply beacon-chain/powchain/service.go]: Fixed. Wasn't sure if we wanted to log a warning or an error here so kept it at warning for now. Lmk if you think this is Error worthy

/healthz returns:
```
*p2p.Service: OK
*powchain.Service: ERROR, genesis state missing, eth1 node out of sync
*attestations.Service: OK
*blockchain.Service: OK
*initialsync.Service: OK
*sync.Service: OK
*rpc.Service: ERROR, syncing
*gateway.Gateway: OK
*prometheus.Service: OK
```

Logs show:

```
[2021-07-02 07:44:19]  WARN powchain: Eth1 node is out of sync and genesis state is missing
[2021-07-02 07:44:24]  INFO initial-sync: Waiting for enough suitable peers before syncing required=3 suitable=2
[2021-07-02 07:44:30]  INFO initial-sync: Processing block batch of size 31 starting from  0xae789b9e... 20129/1628021 - estimated time remaining 288h9m9s blocksPerSecond=1.6 peers=5
[2021-07-02 07:44:34]  WARN powchain: Eth1 node is out of sync and genesis state is missing
^C[2021-07-02 07:44:41]  INFO node: Got interrupt, shutting down...
[2021-07-02 07:44:41]  INFO node: Stopping beacon node
[2021-07-02 07:44:41]  WARN initial-sync: Batch is not processed error=context canceled
```

**Comment by prestonvanloon** [comment]
This is a nice improvement to the error message, but this does not solve #9025. 
The solution for #9025 is that `/healthz` should not return an error when eth1 stalls. 
I think this PR will still cause `/healthz` to return a 500 error with an improved error message.

**Comment by baxterjfinch** [comment]
I see! So it seems I just need to remove L561 which throws the error and just have `/healthz` show that `Eth1 is currently syncing`?

File: beacon-chain/powchain/service.go [hunk 2]
```diff
@@ -545,6 +547,17 @@ func (s *Service) waitForConnection() {
 				}).Info("Connected to eth1 proof-of-work chain")
 				return
 			}
+			genState, genErr := s.cfg.BeaconDB.GenesisState(s.ctx)
```
Comment (terencechain) [inline beacon-chain/powchain/service.go:550]:
```suggestion
			genState, err := s.cfg.BeaconDB.GenesisState(s.ctx)
```

**Review by nisdas — CHANGES_REQUESTED** [review]
This is incorrect, doing it this way would break the ability for a node to build a genesis state from raw deposit logs.

**Pull request closed without merging**